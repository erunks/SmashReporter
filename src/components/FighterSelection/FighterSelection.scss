@import '../../styles/base.scss';

.FighterSelection {
  align-items: center;
  display: flex;
  justify-content: center;
  flex-direction: column;

  .fighterRow:first-child {
    border-top: 2px solid black;
  }
  
  @include breakpoint($sm) {
    grid-column: span 2/2;
    grid-row: 1;
  }

  @include breakpoint($xl) {
    grid-column: 2;
  }
}
