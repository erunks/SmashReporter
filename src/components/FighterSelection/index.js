import React from 'react';
import propTypes from 'prop-types';
import { useRecoilValue } from 'recoil';
import { orderedFighters } from 'recoils/fighter';
import FighterRow from 'components/FighterRow';
import chunk from 'lodash/chunk';

import './FighterSelection.scss';

const FighterSelection = () => {
  const fighters = useRecoilValue(orderedFighters);

  return (
    <div className="FighterSelection">
      {
        chunk(fighters, 12).map(
          (fighterChunk, index) => <FighterRow
              key={`fighterRow${index}`}
              fighterChunk={fighterChunk}
            />
        )
      }
    </div>
  )
};

const {
  arrayOf,
  object
} = propTypes;

FighterSelection.propTypes = {
  initFighters: arrayOf(object)
};

export default FighterSelection;
