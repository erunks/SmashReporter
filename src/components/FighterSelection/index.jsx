import React, { useEffect, useState } from 'react';
import propTypes from 'prop-types';
import chunk from 'lodash/chunk';
import { getFighters } from '../../lib/api';
import FighterRow from '../FighterRow/index.jsx';

import './FighterSelection.scss';

const FighterSelection = ({ initFighters = [{}], ...passThroughProps }) => {
  const [fighters, setFighters] = useState(initFighters);

  useEffect(() => {
    getFighters().then((fighters) => {
      setFighters(fighters);
    });
  }, [])

  const fightersPopulated = () => Object.getOwnPropertyNames(fighters[0]).length

  return (
    <div className="FighterSelection">
      {
        fightersPopulated() && chunk(fighters, 12).map((fighterChunk, index) => <FighterRow
            key={`fighterRow${index}`}
            fighterChunk={fighterChunk}
            {...passThroughProps}
          />)
      }
    </div>
  )
};

const {
  arrayOf,
  func,
  number,
  object
} = propTypes;

FighterSelection.propTypes = {
  initFighters: arrayOf(object),
  selectingPlayer: number.isRequired,
  setPlayerFighter: func.isRequired
};

export default FighterSelection;
