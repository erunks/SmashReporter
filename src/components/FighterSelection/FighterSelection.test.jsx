import React from 'react';
import { mount } from 'enzyme';
import { RecoilRoot } from 'recoil';
import FighterSelection from './index.jsx';

describe('<FighterSelection>', () => {
  const container = mount(
    <RecoilRoot>
      <FighterSelection/>
    </RecoilRoot>
  );

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
