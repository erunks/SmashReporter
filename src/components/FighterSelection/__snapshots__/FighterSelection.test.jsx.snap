// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FighterSelection> should match the snapshot 1`] = `"<div class=\\"FighterSelection\\">0</div>"`;
